src/app/core/services/phase.service.ts

```diff
@@ -85,6 +92,34 @@ export class PhaseService {
     }
   }
 
+  /**
+   * Checks if the necessary repository is available and creates it if the permissions are available.
+   * @param sessionData
+   */
+  verifySessionAvailability(sessionData: SessionData): Observable<boolean> {
+    return this.githubService.isRepositoryPresent(this.phaseRepoOwners[this.currentPhase], sessionData[this.currentPhase]);
+  }
+
+  /**
+   * If a Session is unavailable (because the repository is missing) attempt to create IF it is
+   * the BugReporting Phase
+   * @param sessionData - SessionData containing repository information.
+   * @return - Dummy Observable to give the API sometime to propagate the creation of the new repository since
+   *           the API Call used here does not return any response.
+   */
+  attemptSessionAvailabilityFix(sessionData: SessionData): Observable<any> {
+    if (this.currentPhase !== Phase.phaseBugReporting) {
+      throw new Error('Current Phase\'s Repository has not been opened.');
+    } else if (this.currentPhase === Phase.phaseBugReporting && this.userService.currentUser.role !== UserRole.Student) {
+      throw new Error('Bug-Reporting Phase\'s repository initialisation is only available to Students.');
+    }
```

Comment:
I get a feeling that exceptions are being used for _normal_ flow control.

Reply:
Yes, the initial intention was a means to cut the authentication procedure short should there be an unexpected result from any of the intermediary steps. Also given that error-handling in the app also outputs a useful snackbar with related information, I thought that using errors this way could assist us in addressing issues (especially should they arise during the PE dry run).

I will look into a better way to handle unexpected outputs during the flow of operations as a separate PR 👍